content/geometry/CircleCircleArea.h

```diff
@@ -0,0 +1,21 @@
+/**
+ * Author: Takanori MAEHARA, chilli
+ * Date: 2019-11-03
+ * License: CC0
+ * Source: https://github.com/spaghetti-source/algorithm/blob/master/geometry/_geom.cc#L729
+ * Description: Calculates the area of the intersection of 2 circles
+ * Status:
+ */
+
+template<class P>
+double intersection_area(P c, double cr, P d, double dr) {
+  if (cr < dr) swap(c, d);
+  auto A = [&](double r, double h) {
+    return r*r*acos(h/r)-h*sqrt(r*r-h*h);
+  };
+  auto l = (c - d).dist(), a = (l*l + cr*cr - dr*dr)/(2*l);
+  if (sgn(l - cr - dr) >= 0) return 0; // far away
+  if (sgn(l - cr + dr) <= 0) return M_PI*dr*dr;
+  if (sgn(l - cr) >= 0) return A(cr, a) + A(dr, l-a);
```

Comment:
I'm not a fan of the `sgn`s here, there's no reason why this should be dependent on epsilons.

Reply:
Arguably if you pass in two circles that don't touch, you want it to return absolute 0.

I do think it fails gracefully though, so I guess it's fine to remove it.